filebeat/input/filestream/input.go

```diff
@@ -380,6 +381,15 @@ func (inp *filestream) readFromSource(
 
 		s.Offset += int64(message.Bytes) + int64(message.Offset)
 
+		flags, err := message.Fields.GetValue("log.flags")
+		if err == nil {
+			if flags, ok := flags.([]string); ok {
+				if slices.Contains(flags, "truncated") {
```

Comment:
Can we somehow propagate the metrics registry to message_buffer and increment the counters directly without looking up the flags here? This is a very hot code path, might impact the performance.

Reply:
My understanding is that it is not feasible to move this metric into the message_buffer. First, the metric proposed here (and the overall idea) is to introduce an input-level metric, which, as the name suggests, belongs in the input layer.

Secondly, based on my assessment, the cost of this code is negligible. While slices.Contains is technically unbounded, the log.flags field only holds a small, fixed set of values—currently truncated and multiline. Given this, the overhead can be considered minimal.

The key benefit is that any message setting the truncated flag will automatically increment the metric. Examples of this today include `filestream.message_max_bytes`, `multiline.max_lines`, and the `truncate_fields` processor.

Please let me know your thoughts on this.

Reply:
I'm fine with leaving it as it is as long as we confirm with a benchmark run that there is no significant impact on EPS after this change.

Reply:
Thanks, I triggered a new benchmark run https://buildkite.com/elastic/filebeat-benchmark/builds/455.

This should take a while, I'll get back with the results once it is done.

Reply:
Yeah, looks about the same. Thanks!